### PR TITLE
Add fastpath for restore

### PR DESCRIPTION
**Pull request by ericstj**
This picks up the latest buildtools which includes a task that adds a
fast path for restore.  We will now call that task to do a check up
front to determine if a restore is actually needed before invoking
restore.  This helps the common case where everything is up-to-date
reducing the restore from 5 minutes down to 1 second.

/cc @weshaggard @dagood

### CODE REVIEW

**Comment by dagood** [comment]
The changes look good to me, but it looks like the `NuGet.ProjectModel` version didn't line up.

**Comment by weshaggard** [comment]
@hughbe I believe this change should address https://github.com/dotnet/corefx/issues/10719.

File: build.proj [hunk 1]
```diff
@@ -81,7 +81,12 @@
 
     <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Restoring all packages..." />
 
+    <IsRestoreRequired ProjectJsons="@(ProjectJsonFiles)" PackagesFolder="$(PackagesDir)">
+      <Output TaskParameter="RestoreRequired" PropertyName="RestoreRequired" />
+    </IsRestoreRequired>
+    
     <Exec Command="$(DnuRestoreCommand) @(DnuRestoreDir->'&quot;%(Identity)&quot;', ' ')"
```
Comment (jasonwilliams200OK) [inline build.proj]: Should this be renamed to `DotnetRestoreCommand`? Dnu sounds like a yesterday's news.

Comment (dagood) [inline build.proj]: The name spans a few repos and is used in TFS (last I checked) so it's not a simple rename. It should be renamed, though, and in my opinion to something more generic. (`RestoreCommand`? `ProjectRestoreCommand`?)

**Comment by ericstj** [comment]
I need to pick up a buildtools update for this.  The first build was missing a new dependency that was added.

File: build.proj (hunk 1 above)
Comment (weshaggard) [inline build.proj]: I agree it should be renamed and be more generic. I would actually like to see it moved out of the repo targets and into BuildTools common targets.

**Comment by ericstj** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

File: build.proj (hunk 1 above)
Comment (ericstj) [inline build.proj]: I can do this in a follow up PR.

**Comment by ericstj** [comment]
fixes #10719

**Pull request merged**